src/__tests__/unit/app/api/register-to-vote/route.test.ts

```diff
@@ -0,0 +1,318 @@
+import type { Auth } from '@/services/server/auth/auth';
+import type { User } from '@/model/types/user';
+import type { UserRepository } from '@/services/server/user-repository/user-repository';
+import type { VoterRepository } from '@/services/server/voter-registration-repository/voter-registration';
+import { DateTime } from 'luxon';
+import { UserType } from '@/model/enums/user-type';
+import { AuthError } from '@supabase/supabase-js';
+
+describe('POST', () => {
+  const getActualService = saveActualImplementation(serverContainer, 'get');
+
+  it('takes voter registration information, calls /registertovote api, updates completed task, awards badge, and updates user', async () => {
+    const user: User = {
+      uid: '0',
+      email: 'user@example.com',
+      name: 'User',
+      avatar: '0',
+      type: UserType.Challenger,
+      completedActions: {
+        electionReminders: false,
+        registerToVote: false,
+        sharedChallenge: false,
+      },
+      completedChallenge: false,
+      badges: [],
+      contributedTo: [],
+      challengeEndTimestamp: DateTime.now().plus({ days: 8 }).toUnixInteger(),
+      inviteCode: 'test-invite-code',
+    };
+
+    const insertVoterRegistrationInfo = jest.fn();
+    const awardAndUpdateVoterRegistrationBadgeAndAction = jest.fn();
+
+    const containerSpy = jest
+      .spyOn(serverContainer, 'get')
+      .mockImplementation(key => {
+        if (key.name === SERVER_SERVICE_KEYS.Auth.name) {
+          return Builder<Auth>()
+            .loadSessionUser(() => Promise.resolve(user))
+            .build();
+        } else if (key.name === SERVER_SERVICE_KEYS.UserRepository.name) {
+          return Builder<UserRepository>()
+            .getUserById(() => Promise.resolve(user))
+            .awardAndUpdateVoterRegistrationBadgeAndAction(
+              awardAndUpdateVoterRegistrationBadgeAndAction,
+            )
+            .build();
+        } else if (key.name === SERVER_SERVICE_KEYS.VoterRepository.name) {
+          return Builder<VoterRepository>()
+            .insertVoterRegistrationInfo(insertVoterRegistrationInfo)
+            .build();
+        }
+        return getActualService(key);
+      });
+
+    const registerBody = {
+      user_id: '0',
+      state: 'FL',
+      city: 'Davie',
+      street: '2161 SW 152 Ter',
+      name_first: 'John',
+      name_last: 'Doe',
+      dob: '09/20/2003',
+      zip: '33027',
+      email: 'test@me.com',
+      citizen: 'yes',
+      eighteenPlus: 'yes',
+      party: 'Democrat',
+      idNumber: '123',
+    };
+
+    const request = new NextRequest(
+      'https://challenge.8by8.us/register-to-vote',
+      {
+        method: 'POST',
+        body: JSON.stringify(registerBody),
+      },
+    );
+
+    const response = await POST(request);
+    expect(response.status).toBe(200);
+    containerSpy.mockRestore();
+  }, 100_000);
+
+  it('calls the route with a null user', async () => {
+    const containerSpy = jest
```

Comment:
I think the string describing your assertion here could be more specific. For instance "It returns a Response with a status of 401 when the user is signed out." I will update this when I perform some merges today, but please remember this for the future :)